### DISCUSSION TITLE
ElastAlert: Rule Mismatch + Suricata: Rule Mismatch

### DISCUSSION THREAD

**Discussion by mohasmr** [Security-Onion-Solutions/securityonion#13238] [2.4] [answered]
### Version

2.4.70

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

airgap

### Hardware Specs

Meets minimum requirements

### CPU

8

### RAM

32

### Storage for /

300GB

### Storage for /nsm

204 GB

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

i Runned full update only changes the status to **pending** and then back to **rule mismatch**  for elastalert and suricata when it finishes and strelka is ok 
i also tried to run **salt-call state.highstate**  from the discussion #13112
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/135219197/8b9210d1-413f-44bd-9564-5c37f630d3c2)

there is no fail here 

but it changes  to **pending** and then back to **rule mismatch** 

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by defensivedepth** [comment]
Do you have any custom rules?

Also, click on the "Rule Mismatch", and filter for the "Integrity Check Report" log - this will give more specific details about what has failed.

**Comment by mohasmr** [comment]
this is some details about integrity check report
{"fields":{"deployedButNotEnabled":[],"detectionEngine":"suricata","enabledButNotDeployed":["2005522"],"intCheckId":"b0f2d2c1-b434-4271-915c-d5eaf2777aeb"},"level":"info","timestamp":"2024-06-21T07:16:16.842641648Z","message":"integrity check report"}

{"fields":{"deployedButNotEnabled":["2053760","2053781","2053750","2053777","2053767","2053779","2053768","2053756","2053749","2053773","2053778","2053776","2053752","2053751","2053771","2053759","2053758","2053772","2053780","2053755","2053783","2053765","2053762","2053770","2053754","2053757","2053774","2053775","2053769","2053764","2053761","2053766","2053753","2053782","2053763"],"detectionEngine":"suricata","enabledButNotDeployed":["2053723","2522791","2053724","2005522","2522790","2522789","2525011"],"intCheckId":"d141c0c2-4c64-495a-95bf-12a9c8e3ec2b"},"level":"info","timestamp":"2024-06-21T07:14:36.827987021Z","message":"integrity check report"}


{"fields":{"deployedButNotEnabled":["2053766","2053762","2053765","2053783","2053774","2053763","2053759","2053750","2053761","2053771","2053780","2053755","2053764","2053756","2053749","2053778","2053781","2053779","2053768","2053782","2053757","2053767","2053752","2053777","2053775","2053772","2053776","2053769","2053770","2053773","2053760","2053751","2053753","2053754","2053758"],"detectionEngine":"suricata","enabledButNotDeployed":["2525011","2053723","2522789","2005522","2053724","2522791","2522790"],"intCheckId":"d2f63ae1-a4ef-402d-bd70-c11e332ac1b7"},"level":"info","timestamp":"2024-06-21T07:04:29.945099042Z","message":"integrity check report"}




i did'nt add custom rules,the rules i add it was from detection section in the GUI

**Reply by defensivedepth** [reply]
Ok, then this is a known bug that will be fixed in the next release.

**Reply by defensivedepth** [reply, marked as answer]
@mohasmr 2.4.80 was released yesterday and has a number of fixes for Detections. I would suggest upgrading and seeing if that fixes the issue.

**Reply by jasondad36** [reply]
Unfortunately, I have the same issue, and it started when upgrading form 2.4.70 to 2.4.80.

soc.fields.enabledButNotDeployed: 2012078,2031610,2051480,2012079,2051473,2016150,2018959,2021788,2052865,2012080,2040272,2040271,2046211,2016149,2052866,2033077

soc.fields.deployedButNotEnabled: 340a090b-c4e9-412e-bb36-b4b16fe96f9b

Edit:
I was able to get it fixed by just going into the detection tab, using the top search bar to search for the SID number referenced in the reports and disabled them. Now its all green. I am not sure why those SIDS do not get deployed when enabled though.

**Reply by Carlos-mb** [reply]
Hi @jasondad36 

How do you get the integrity check report? I haven't find info about it. 

Thanks!

**Reply by jasondad36** [reply]
In detections, in the top right corner where it says "ElastAlert Rule: Mismatch", Click that. It will open up the hunt section. In the middle section, there should be an event.action that says: integrity check failed. Click that and select "only". Then in the bottom section, you want to look for the events that are "integrity check failed." That will provide the lists of issue signatures.

**Reply by Carlos-mb** [reply]
@jasondad36 It's incredible. I was so many hours trying to know how to do that. Thank you for your explanation.

**Reply by dougburks** [reply]
This is mentioned at https://docs.securityonion.net/en/2.4/detections.html#rule-engine-status:
```
Clicking the status text will navigate to Hunt and attempt to find related logs.
```

In the next version, that text will be highlighted as a tip and provide more detail:
![image](https://github.com/user-attachments/assets/d5092a0e-0f6f-483d-ae95-949366f0334c)

**Reply by Carlos-mb** [reply]
Thanks @dougburks, I've already read it, now I understand the meaning and what exactly I should do.

**Comment by mohasmr** [comment]
so is there any solution to reset the manager without losing the sensor and search node or a solution to reset the rules to the default like pre installation

**Reply by defensivedepth** [reply]
@mohasmr Because this is a new issue, please start a new Discussion

**Comment by mohasmr** [comment]
the issue resolved after the upgrade
